src/main/scala/abalone/Pos.scala

```diff
@@ -190,7 +190,41 @@ case class Pos private (index: Int) extends AnyVal {
   def right: Option[Pos]     = Pos.at(file.index + 1, rank.index)
   def downRight: Option[Pos] = Pos.at(file.index, rank.index - 1)
   def upRight: Option[Pos]   = Pos.at(file.index + 1, rank.index + 1)
-  def neighbours: List[Option[Pos]] = List(left, downLeft, upLeft, right, downRight, upRight)
+
+  def neighbours: List[Option[Pos]] = List(left, upLeft, upRight, right, downRight, downLeft)
+  def sideMovesDirsFromDir(dir: Option[String]): (Option[Pos], Option[Pos]) = Map(
```

Comment:
Same as below, its unnecessary that you are creating a map and then getting an item from it, just you a match on the dir and have a default case returning none.
same for function below.